apps/extension/src/senders/validate.ts

```diff
@@ -34,7 +34,13 @@ export const assertValidSender = (sender?: chrome.runtime.MessageSender) => {
   if (parsedOrigin.origin !== sender.origin) {
     throw new Error('Sender origin is invalid');
   }
-  if (!(parsedOrigin.protocol in ValidProtocol)) {
+
+  if (
+    !(
+      parsedOrigin.protocol in ValidProtocol ||
+      (globalThis.__DEV__ && isHttpLocalhost(parsedOrigin))
+    )
+  ) {
```

Comment:
todo: we should have a test for this in validate.test.ts. Also, could you check that test suite to ensure we've covered all cases for this function?

Reply:
@grod220 thanks for pointing this out. I should pay more attention to the tests